File: Rust/Medium/Accepted/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold_6.rs
impl Solution {
    pub fn num_of_subarrays(arr: Vec<i32>, k: i32, threshold: i32) -> i32 {
        
        let mut sum = arr[0..k as usize].iter().sum::<i32>();
        let mut count = if sum / k >= threshold { 1 } else { 0 };
        for i in 0..arr.len() - k as usize {
            sum = sum - arr[i] + arr[i + k as usize];
            if sum / k >= threshold {
                count += 1;
            }
        }
        count
    }
}

File: C++/Medium/Accepted/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold_6.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        
        int sum = 0;
        int n = arr.size();
        int res = 0;
        for (int i = 0; i < n; ++i) {
            sum += arr[i];
            if (i >= k) sum -= arr[i - k];
            if (i >= k - 1) {
                if (sum / k >= threshold) ++res;
            }
        }
        return res;
    }
};